build-scripts/lerna-publish.js

```diff
@@ -0,0 +1,144 @@
+const createCmd = require('@lerna/create/command');
+const diffCmd = require('@lerna/diff/command');
+const execCmd = require('@lerna/exec/command');
+const importCmd = require('@lerna/import/command');
+const initCmd = require('@lerna/init/command');
+const linkCmd = require('@lerna/link/command');
+const listCmd = require('@lerna/list/command');
+const publishCmd = require('@lerna/publish/command');
+const runCmd = require('@lerna/run/command');
+const versionCmd = require('@lerna/version/command');
+const Semver = require('semver');
+const pkg = require('lerna/package.json');
+const { PublishCommand } = require('@lerna/publish');
+const fs = require('fs');
+const path = require('path');
+
+function factory(argv) {
+  return new TravisPublishCommand(argv);
+}
+
+/** Override methods in the main publication command class to return the full set of packages for publication */
+class TravisPublishCommand extends PublishCommand {
+  findVersionedUpdates() {
+    let chain = Promise.resolve();
+
+    if (this.options.bump === "from-git") {
+      chain = chain.then(() => this.detectFromGit());
+    } else if (this.options.canary) {
+      chain = chain.then(() => this.detectCanaryVersions());
+    } else {
+      // If no version bump was specified, check to ensure the working directory is clean
+      // and then attempt to publish all packages.
+      chain = chain
+          .then(() => this.verifyWorkingTreeClean())
+          .then(() => Array.from(this.packageGraph.values()))
+          .then(updates => {
+            const updatesVersions = updates.map(({ pkg }) => [pkg.name, pkg.version]);
+
+            return {
+              updates,
+              updatesVersions,
+              needsConfirmation: true,
+            };
+          });
+    }
+
+    return chain;
+  }
+}
+
+// New command meta data
+const travisPublishCmd = Object.assign({}, publishCmd, {
+  command: 'publish-travis [bump]',
+  describe: 'Publish all packages in the current project',
+  handler: factory
+});
+
+// Setup the custom cli
+function main(argv) {
+  // For lerna publication to work, the NPM token must be stored in the .npmrc file in the user home directory
+  if (process.env.TRAVIS && process.env.NPM_TOKEN) {
+    fs.writeFileSync(
+        path.resolve(process.env.HOME, '.npmrc'),
+        `//registry.npmjs.org/:_authToken=${process.env.NPM_TOKEN}\n`,
+        'utf8');
+  }
+
+  const context = {
+    lernaVersion: pkg.version,
+  };
+
+  return cli()
+      .command(addCmd)
+      .command(bootstrapCmd)
+      .command(changedCmd)
+      .command(cleanCmd)
+      .command(createCmd)
+      .command(diffCmd)
+      .command(execCmd)
+      .command(importCmd)
+      .command(initCmd)
+      .command(linkCmd)
+      .command(listCmd)
+      .command(publishCmd)
+      .command(runCmd)
+      .command(versionCmd)
+      .command(travisPublishCmd)
+      .parse(argv, context);
+}
+
+if (/publish/.test(process.argv[2])) {
```

Comment:
What are the valid first args here?
Are we really checking specifically for "publish", or any string with "publish" in it?

Reply:
We're looking for either `publish` or `publish-travis`. I added a comment.